junit5/src/main/java/com/linecorp/armeria/testing/junit5/server/ServerExtension.java

```diff
@@ -51,9 +55,11 @@ protected ServerExtension() {
      *                  {@code false} if the {@link Server} should start when a user calls {@link #start()}.
      */
     protected ServerExtension(boolean autoStart) {
+        contextCaptor = new ServiceRequestContextCaptor();
         delegate = new ServerRuleDelegate(autoStart) {
             @Override
             public void configure(ServerBuilder sb) throws Exception {
+                sb.decorator(contextCaptor.decorator(ServerExtension.this::shouldCapture));
                 ServerExtension.this.configure(sb);
```

Comment:
We need to specify the decorator lastly so that the context is captured as soon as the request is received, because otherwise a user's decorator may intercept or drop it.
```suggestion
                ServerExtension.this.configure(sb);
                sb.decorator(contextCaptor.decorator(ServerExtension.this::shouldCapture));
```